app/agreement/create.js

```diff
@@ -9,7 +9,8 @@ const AGREEMENT = {
   organisation: {},
```

Comment:
Is this something to be done for each of the sub caches within the session cache? Makes it very easy to know the defaults for each component making edge/test cases less (unless a user deleted cache parts) -- e.g. seeing that funding defaults to an empty array helps with reducing `cache.agreement?.funding ?? []` and means we can use the standard `funding.length` for boolean checks

Reply:
Yes I think that makes sense as an approach.  Having the data structure created once consistently and then updated throughout the journey makes sense and behaviour is more predictable as you say.

Currently we're only doing this for agreement element.  I think it does make sense to think about how we could do the same for the rest of it.